processor/groupbyattrsprocessor/README.md

```diff
@@ -83,6 +86,64 @@ Notes:
 * The specified "grouping" attributes that are set on the new *Resources* are also **removed** from the metric *DataPoints*
 * While not shown in the above example, the processor also merges collections of records under matching InstrumentationLibrary
 
+### Compaction
+
+In some cases, the data might come in single requests to the collector and even after batching there might be multiple duplicated ResourceSpans/ResourceLogs/ResourceMetrics objects, which leads to additional memory consumption and increased processing costs. As a remedy, `groupbyattrs` processor might be used to compact the data which has matching Resource and InstrumentationLibrary properties.
+
+For example, consider the following input:
+
+```go
+Resource {host.name="localhost"}
+  InstumentationLibrary {name="MyLibrary"}
+  Spans
+    Span {span_id=1, ...}
+  InstumentationLibrary {name="OtherLibrary"}
+  Spans
+    Span {span_id=2, ...}
+    
+Resource {host.name="localhost"}
+  InstumentationLibrary {name="MyLibrary"}
+  Spans
+    Span {span_id=3, ...}
+    
+Resource {host.name="localhost"}
+  InstumentationLibrary {name="MyLibrary"}
+  Spans
+    Span {span_id=4, ...}
+    
+Resource {host.name="otherhost"}
+  InstumentationLibrary {name="MyLibrary"}
+  Spans
+    Span {span_id=5, ...}
+```
+
+With the below configuration, the **groupbyattrs** will re-associate the spans with matching Resource and InstrumentationLibrary
+
+```yaml
+processors:
+  groupbyattrs:
```

Comment:
When should it be used? Before the batching processor? After? Always, or only when there's a specific characteristic?